lib/client.js

```diff
@@ -598,18 +597,23 @@ class Client {
             this.error(500, err);
           }
           application.logException(err);
-          return;
         }
         callback();
       }
     };
 
-    try {
-      fn(this, done);
-    } catch (err) {
-      this.error(500, err);
-      application.logException(err);
-      callback();
+    if (fn.constructor.name === 'AsyncFunction') {
+      try {
```

Comment:
I think we can just use `Promise` API here:
```suggestion
      fn(this).then(result => {
        done(null, result);
      }, err => {
        done(err);
      });
```
This will also allow to make the method `executeFunction()` non-async, which in turn will allow to use it with Node.js versions where `async` is not yet available.